services/venice-controller/src/main/java/com/linkedin/venice/controller/ZkHelixAdminClient.java

```diff
@@ -328,4 +325,35 @@ public void close() {
   public void addInstanceTag(String clusterName, String instanceName, String tag) {
     helixAdmin.addInstanceTag(clusterName, instanceName, tag);
   }
+
+  public void setCloudConfig(VeniceControllerClusterConfig config, boolean isControllerCluster) {
```

Comment:
I like how you've abstracted both of these. But instead of passing `isControllerCluster` can we check in the caller if the cloud config should be set and only then call `setCloudConfig`?

So, that'd be like these two:
```java
if (commonConfig.isControllerClusterHelixCloudEnabled) {
    setCloudConfig(commonConfig);
}
```

```java
if (config.isControllerStorageClusterHelixCloudEnabled) {
    setCloudConfig(config);
}
```

Passing a boolean and handling it two different ways based on knowledge of when this boolean is set kind of breaks abstraction principles